_drafts/2017-09-01-first-review-experiences.md

```diff
@@ -0,0 +1,137 @@
+---
+name: first-review-experiences
+layout: post_discourse
+title: Experiences as a first time rOpenSci package reviewer
+date: 2017-09-01
+authors:
+  - name: Verena Haunschmid
+    url: http://blog.haunschmid.name
+tags:
+- R
+- software
+- review
+- ropensci
+- onboarding
+---
+
+It all started January 26^th^ this year when I signed up to volunteer as
+a reviewer for R packages submitted to rOpenSci. My main motivation for
+wanting to do package reviews was to learn something new and to
+contribute to the R open source community.
+
+Three months later I was contacted by [Ma�lle Salmon](https://twitter.com/ma\_salmon) asking whether I was interested in
+reviewing the R package ['patentsview'](https://github.com/ropensci/patentsview) for rOpenSci. And yes, I
+was! To be honest I was a little bit thrilled.
+
+The packages are submitted for review to rOpenSci via an issue to their
+github repository and also the reviews happen there. So you can check out all
+previous package submissions and reviews. With all the information you
+get from rOpenSci and also the help from the editor it is straight
+forward to do the package review. Before I started my review I read the
+reviewer guides (links below) and checked out a few of the existing
+reviews. I installed the package 'patentsview' from github and also
+downloaded the source code so I could check out how it was implemented.
+There are several R packages that helped me get started with my review,
+e.g. ['devtools'](https://github.com/hadley/devtools) and
+['goodpractice'](https://github.com/MangoTheCat/goodpractice). These
+packages can also help you when you start writing your own packages. An
+example for a very useful method is 'devtools::spell\_check()', which
+performs a spell check on the package description and on manual pages.
+At the beginning I had an issue with 'goodpractice::gp()' but Ma�lle Salmon 
+(the editor) helped me resolve it. After that I started my review
+by testing core functionality of the package by running the examples
+that were mentioned in the README of the package. I think this is a good
+starting point because you get a feeling of what the author wants to
+achieve with the package. Later on during the review I came up with my
+own queries (side note: this R package interacts with an API from which
+you can query patents). During the review I used to switch between 
+writing queries like a normal user of the package
+would do and checking the code. When I saw something in the code that
+wasn't quite clear to me or looked wrong I went back to writing new
+queries to check whether the behavior of the methods was as expected.
+With this approach I was able to give feedback to the package author
+which led to the inclusion of an additional unit test, a helper function
+that makes the package easier to use, clarification of an error message
+and an improved documentation. You can find the review I did [here](https://github.com/ropensci/onboarding/issues/112).
```

Comment:
Can you find a place in your post to refer to this new blog post by our editorial team? [How rOpenSci uses Code Review to Promote Reproducible Science](https://www.numfocus.org/blog/how-ropensci-uses-code-review-to-promote-reproducible-science/).

and at the end would you consider "inviting" people to join our Sep 13 [Community Call on rOpenSci software review and onboarding](https://ropensci.org/blog/blog/2017/08/31/comm-call-v14)? Maybe even say something like "join me on ..." or "I will be joining ...". It's always great in our asynchronous online community to help people see that there are opportunities to connect with real humans!